### PR TITLE
Add support for RFC9073 & RFC9074

### PR DESCRIPTION
**Pull request by ksmurchison**
Since we don't currently run restriction checks on anything but children of VCALENDAR, I haven't bothered to add any of the new VALARM property restrictions.

I will open another PR to support restriction checks deeper than one level.

### CODE REVIEW

**Comment by winterz** [comment]
please add these newly supported RFCS to the supported RFC list in the README.md, doc/UsingLibical.md and doc/Mainpage.dox

maybe we should have a separate .md file listing the RFCs we support.
but that's a different thread

File: src/libical/icalrestriction.c.in [hunk 1]
```diff
@@ -99,109 +99,93 @@ int icalrestriction_compare(icalrestriction_kind restr, int count)
...
-    }
-
-    return 0;
-}
-
-static const char *icalrestriction_may_be_comp_need_process(const icalrestriction_record *
-                                                            rec, icalcomponent *comp,
-                                                            icalproperty *prop)
-{
     icalproperty_status stat = icalproperty_get_status(prop);
```
Comment (winterz) [inline src/libical/icalrestriction.c.in:107]: we need to have the` icalproperty_status stat` declaration above the `if (!prop) return 0;` else ancient C compilers complain

**Comment by ksmurchison** [comment]
@winterz What are the AppVeyor complaints?

**Comment by winterz** [comment]
I'm seeing regression failures at libical/src/test/regression-classify.c:130

        Failed 7/1064 tests, 99% okay

        Failed tests:

          46/823 46/825 46/827 46/829 46/835 46/847 46/849

**Comment by winterz** [comment]
> @winterz What are the AppVeyor complaints?

looks like the regression test is failing, as I already reported by my local testing

**Comment by winterz** [comment]
looks great now.
one last little thing -> please mention this in the 3.1.0 section in ReleaseNotes

**Review by ksuther — APPROVED** [review]
Looks good to me. I wasn't aware of 9073 and 9074, so thanks for pointing out the existence of those.

**Pull request merged**